external-crates/move/crates/move-compiler/tests/move_check/expansion/invalid_unpack_assign_lhs_other_value.exp

```diff
@@ -7,3 +7,23 @@ error[E01002]: unexpected token
   │           Unexpected '{'
   │           Expected ';'
 
+error[E03003]: unbound module member
+  ┌─ tests/move_check/expansion/invalid_unpack_assign_lhs_other_value.move:5:9
+  │
+5 │         foo() = 0;
+  │         ^^^ Invalid module access. Unbound struct 'foo' in module '0x42::M'
+
+error[E13001]: feature is not supported in specified edition
+  ┌─ tests/move_check/expansion/invalid_unpack_assign_lhs_other_value.move:5:9
+  │
+5 │         foo() = 0;
+  │         ^^^^^ Positional fields are not supported by current edition 'legacy', only '2024.alpha' and '2024.beta' support this feature
+  │
+  = You can update the edition in the 'Move.toml', or via command line flag if invoking the compiler directly.
+
+error[E01009]: invalid assignment
+  ┌─ tests/move_check/expansion/invalid_unpack_assign_lhs_other_value.move:7:9
+  │
+7 │         foo().bar() = 0;
+  │         ^^^^^^^^^^^ Invalid assignment syntax. Expected: a local, a field write, or a deconstructing assignment
```

Comment:
These errors indicate we need to do something smarter in expansion / naming. Not on this diff, but a high-priority follow-up.